bitcoin_tools/analysis/status/utils.py

```diff
@@ -377,6 +385,48 @@ def decode_utxo_v08_v014(utxo):
 
     return {'version': version, 'coinbase': coinbase, 'outs': outs, 'height': height}
 
+def decompress_script(script_type,script_bytes):
+    """ Takes CScript as stored in leveldb and returns it in uncompressed form
+    (de)compression scheme is defined in bitcoin/src/compressor.cpp
+
+    :param script_type: first byte of script data (out_type in decode_utxo)
+    :type script_type: int
+    :param script_bytes: raw script bytes hexlified (data in decode_utxo)
+    :type script_bytes: str
+    :return: the decompressed CScript
+    :rtype: str
+    """
+
+    data = unhexlify(script_bytes)
+    if script_type == 0:
+        assert len(data) == 20
+        data = OP_DUP + OP_HASH160 + chr(20) + data + \
+            OP_EQUALVERIFY + OP_CHECKSIG
+
+    elif script_type == 1:
+        assert len(data) == 20
+        data = OP_HASH160 + chr(20) + data + OP_EQUAL
+
+    elif script_type == 2 or script_type == 3:
+        data = data[1:]
+        assert len(data) == 32
+        data = chr(33) + script_type + data + OP_CHECKSIG
+
+    elif script_type == 4 or script_type == 5:
+        data = data[1:]
+        assert len(data) == 32
+
+        comp_pubkey = chr(script_type - 2) + data
+        pubkey = secp256k1.PublicKey(
+            comp_pubkey, raw=True).serialize(compressed=False)
+
+        data = chr(65) + pubkey + OP_CHECKSIG
+
+    else:
+        assert len(data) == script_type - 6
```

Comment:
Don't use 6 as magic number, NSPECIALSCRIPTS is defined with that purpose. Actually that check should be `(out_type - NSPECIALSCRIPTS) * 2`since the comparison is being made with a `str` data, so size is doubled.